http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -317,6 +331,24 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
           clusterQuery.updateClusterStatus(existingCluster.id, ClusterStatus.Updating, now).transaction.void
         } else IO.unit
 
+        //maybeMasterMachineTypeChanged will throw an error if the appropriate flag hasn't been set to allow special update transitions
+        shouldFollowup = masterMachineTypeChanged.map(result => result.shouldFollowup).getOrElse(false)
+        _ <- if (shouldFollowup) {
+          //we do not support follow-up transitions when the cluster is set to an updating status
+          if (!shouldUpdate.combineAll) {
+            val action = masterMachineTypeChanged.map(result => result.followupAction).getOrElse(Noop(None))
+            logger.info(s"detected follow-up action necessary: ${action}")
+            handleClusterTransition(existingCluster, action)
+          } else {
+            logger.warn(
+              "A user tried to resize the cluster and master machine type at the same time, which is not supported. Only the cluster will be resized."
```

Comment:
I'm beginning to think this should be an exception, as it could be really confusing for a user. Thoughts?

Reply:
Would be nice to have the validation in UI.

but I think in API, we should also notify user. Not "throwing" an exception though, just do `IO.raiseError(new someBadRequestException)`

Reply:
Hm… I tend to think that returning an error (409 Conflict?) is cleaner than silently ignoring the request to change the machine type.

I do agree that this is not expected to be a common transition, so not worth the complexity to unify the `Updating` status with the stop-start transition.